modules/syslog.lua

```diff
@@ -30,12 +36,15 @@ local msg_any         = octet^0
 local msg_utf8        = bom * utf8_string
 local msg             = msg_utf8 + msg_any
 local hostname        = nilvalue + printusascii^-255
-local sd_name         = printusascii^-32
-local param_name      = sd_name
-local param_value     = utf8_string
-local sd_param        = param_name * '="' * param_value * '"'
+local sd_name         = (printusascii - l.S'=]" ')^-32
+local param_name      = l.C(sd_name)
+local esc             = l.P'\\'
+local param_value_esc = l.S'"\\]'
+local param_value     = l.C(((octet - param_value_esc) + (esc * octet))^0) / unescape_param_value
 local sd_id           = sd_name
```

Comment:
We may want to include the sd_id in the message also for easier filtering of structured data types